### PR TITLE
Tests for class_load_container

### PR DESCRIPTION
**Pull request by NormannK**
- translate class_load_container.py
- add type hints
- removed Numpy (we converted it right away after generating back)
- created test for class

should have been a 2nd PR:
Renamed the to_dict() function to return_dict() to avoid conflict with Pandas' to_dict() method and to prevent confusion with similarly named functions like to_list().
(i learned from it)

### CODE REVIEW

**Comment by drbacke** [comment]
@NormannK Could you rebase / merge the conflicts please? Then I can test it.

File: src/akkudoktoreos/class_akku.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def __init__(
         self.min_soc_wh = (self.min_soc_prozent / 100) * self.kapazitaet_wh
         self.max_soc_wh = (self.max_soc_prozent / 100) * self.kapazitaet_wh
 
-    def to_dict(self):
+    def return_dict(self):
```
Comment (fortrieb) [inline src/akkudoktoreos/class_akku.py:34]: We should stick to methods names used in Python std libs (`to_dict`).

File: src/akkudoktoreosserver/flask_server.py [hunk 2]
```diff
@@ -165,9 +164,9 @@ def flask_gesamtlast_simple():
         # leistung_wp = wp.simulate_24h(temperature_forecast)  # Simulate heat pump load for 24 hours
         # gesamtlast.hinzufuegen("Heatpump", leistung_wp)  # Add heat pump load to total load calculation
 
-        last = gesamtlast.gesamtlast_berechnen()  # Calculate total load
-        print(last)  # Output total load
-        return jsonify(last.tolist())  # Return total load as JSON
+        last = gesamtlast.calculate_total_load()  # Calculate total load
+        print(last)
```
Comment (fortrieb) [inline src/akkudoktoreosserver/flask_server.py:168]: Using `logging.debug` instead of `print`?

**Review by fortrieb — CHANGES_REQUESTED** [review]

File: src/akkudoktoreosserver/flask_server.py (hunk 2 above)
Comment (NormannK) [reply src/akkudoktoreosserver/flask_server.py:168]: sure but my intention was not to clean up the flask_server for now. Lets make that a new PR

**Comment by drbacke** [comment]
Please re-run all tests

**Comment by NormannK** [comment]
done. no changes needed. please some feedback on the change request

File: src/akkudoktoreos/class_akku.py (hunk 1 above)
Comment (drbacke) [reply src/akkudoktoreos/class_akku.py:34]: yes please, return is uncommon

File: src/akkudoktoreos/class_load_container.py [hunk 3]
```diff
@@ -1,39 +1,47 @@
...
-        if not self.lasten:
-            return []
+        if not self.loads:
+            return []  # Return empty list if no loads are present
 
-        # Assumption: All load arrays have the same length
-        stunden = len(next(iter(self.lasten.values())))
-        gesamtlast_array = [0] * stunden
+        # Initialize the total load array with zeros
+        LoadAggregator_array = [0.0] * self.prediction_hours
```
Comment (drbacke) [inline src/akkudoktoreos/class_load_container.py:40]: I'm not sure if this is sensfull. The Load Container is now dependend on prediction_hours. I think it's not until now.

File: src/akkudoktoreos/class_load_container.py [hunk 4]
```diff
@@ -1,39 +1,47 @@
...
 
-        for last_array in self.lasten.values():
-            gesamtlast_array = [
-                gesamtlast + stundenlast
-                for gesamtlast, stundenlast in zip(gesamtlast_array, last_array)
-            ]
+        # Sum loads from all sources
+        for last_array in self.loads.values():
+            for i in range(self.prediction_hours):
```
Comment (drbacke) [inline src/akkudoktoreos/class_load_container.py:44]: I'm not sure if this is sensfull. The Load Container is now dependend on prediction_hours. I think it's not until now.

**Review by drbacke — CHANGES_REQUESTED** [review]
Some small changes requested.

File: src/akkudoktoreos/class_akku.py (hunk 1 above)
Comment (NormannK) [reply src/akkudoktoreos/class_akku.py]: removed

File: src/akkudoktoreos/class_load_container.py (hunk 3 above)
Comment (NormannK) [reply src/akkudoktoreos/class_load_container.py:40]: the add_load() function in only called in the flask_server. two times to add the household load to total load calculation.
in the first case its using `future_predictions = adjuster.predict_next_hours(prediction_hours) ` as a data source which is based on `prediction_hours`. So my assumption was to make sure we add always the same amount. But i do get the point. We might want to use the function to add only some parts to the data and we don't want to add all. in that case we would have to create an array which only have the data we want to add. I guess thats not needed. ill revert it and make a small change to how the amount of hours is calculated so its imo better to read.

File: src/akkudoktoreos/class_load_container.py (hunk 4 above)
Comment (NormannK) [reply src/akkudoktoreos/class_load_container.py]: removed prediction_hours hers. also used numpy again to make it fast.

**Comment by NormannK** [comment]
@drbacke changed as requested. see change request comments for details.

**Pull request closed without merging**